pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -105,7 +111,7 @@ cdef value_count_{{dtype}}(const {{dtype}}_t[:] values, bint dropna, const uint8
     cdef:
         int64_t[::1] result_counts = np.empty(table.size + na_add, dtype=np.int64)
 
-    for i in range(table.size):
+    for i in range(<Py_ssize_t>table.size):  # cast required for 32 bit builds
```

Comment:
We alternately could change `table.size` to be of type `Py_ssize_t` but leaving that to a separate exercise